caserver/caserver.go

```diff
@@ -406,6 +406,41 @@ func home(w http.ResponseWriter, r *http.Request) {
 	w.Write([]byte("Linaro CA HTTP server\n"))
 }
 
+// X509 Certificate copy handler
+func ccGet(w http.ResponseWriter, r *http.Request) {
+	pathParams := mux.Vars(r)
+
+	// Parse serial number in request
+	serialNumber, ok := pathParams["serial"]
+	ser := new(big.Int)
+	ser, ok = ser.SetString(serialNumber, 10)
+	if !ok {
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte(`{"error": "invalid request"}`))
+		return
+	}
+
+	cert, err := db.GetCertBySerial(ser)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte(`{"error": "invalid serial number"}`))
+		fmt.Println("cs:", err)
+		return
+	}
+	
+	// Convert DER output to PEM
+	pemout := pem.EncodeToMemory(&pem.Block{Type: "CERTIFICATE", Bytes: cert})	
+    
+	w.Header().Set("Content-Type", "application/json")
```

Comment:
As mentionned, we also need to handle requests and responses for `application/cbor`